src/legacy/core_plugins/console/public/np_ready/application/containers/editor/legacy/console_editor/editor.tsx

```diff
@@ -81,10 +86,54 @@ function EditorUI() {
   useEffect(() => {
     editorInstanceRef.current = senseEditor.create(editorRef.current!);
 
-    const { content: text } = history.getSavedEditorState() || {
-      content: DEFAULT_INPUT_VALUE,
+    const getQueryParams = () => {
+      const [, queryString] = window.location.hash.split('?');
+      return qs.parse(queryString || '');
     };
-    editorInstanceRef.current.update(text);
+
+    const onHashChange = debounce(() => {
+      const qParams = getQueryParams();
+      if (!qParams || !qParams.load_from) {
+        return;
+      }
+      const sourceLocation: string = qParams.load_from;
+      if (/^https?:\/\//.test(sourceLocation)) {
+        const loadFrom: Record<string, any> = {
+          url: sourceLocation,
+          // Having dataType here is required as it doesn't allow jQuery to `eval` content
+          // coming from the external source thereby preventing XSS attack.
+          dataType: 'text',
+          kbnXsrfToken: false,
+        };
+
+        if (/https?:\/\/api\.github\.com/.test(sourceLocation)) {
+          loadFrom.headers = { Accept: 'application/vnd.github.v3.raw' };
```

Comment:
What use case does this support?

Reply:
This was copied from the previous behaviour, perhaps there is specific case for loading from github that required an additional header? I thought I would keep the legacy behaviour unchanged until we decide to officially deprecate load from behaviour or revisit it in future.

Reply:
Sounds good, thanks for the explanation!